### ISSUE TITLE
Native support for jsonnet input files

### ISSUE THREAD

**Issue by josdotso** [roboll/helmfile#814]
Similar to how DroneCI supports jsonnet: https://docs.drone.io/extend/config/jsonnet/

**Comment by mumoshu** [comment]
Yeah, I DO want this as I believe Jsonnet is much better than go templates for declaring complex setups.

Jsonnet code can be a little harder to debug/review/audit/etc so I'm thinking that it would play nicely with `helmfile build -f helmfile.jsonnet` that will produce a "rendered" helmfile.yaml from Jsonnet (#780 )

Happy to review any contribution related to this feature.

Implementation-wise, the only thing I'm unsure is that how to add custom functions to Jsonnet, like `requiredEnv`, `exec`, and helmfile values that is missing in the Jsonnet stdlib.
